src/gui/components/TextInput.zig

```diff
@@ -148,22 +152,63 @@ pub fn select(self: *TextInput) void {
 	self.pressed = false;
 	self.selectionStart = null;
 	if(self.cursor == null)
-		self.cursor = @intCast(self.currentString.items.len);
+		self.cursor = @intCast(self.getBufferLen());
 }
 
 pub fn deselect(self: *TextInput) void {
 	self.cursor = null;
 	self.selectionStart = null;
 }
 
+fn cursorPosFromObfuscated(self: *const TextInput) u32 {
+	var iter: std.unicode.Utf8Iterator = .{.bytes = self.currentString.items, .i = 0};
+	return @intCast(iter.peek(self.cursor.?/obfuscationChar.len).len);
+}
+
+fn cursorPosToObfuscated(self: *const TextInput) u32 {
+	const newCursor = (std.unicode.utf8CountCodepoints(self.currentString.items[0..self.cursor.?]) catch 0)*obfuscationChar.len;
+	return @intCast(newCursor);
+}
+
+fn getBufferLen(self: *const TextInput) usize {
+	return if(self.obfuscated) self.textBuffer.glyphs.len*obfuscationChar.len else self.currentString.items.len;
+}
+
+pub fn obfuscate(self: *TextInput) void {
```

Comment:
I think this should be a static option used on construction. Allowing to change this dynamically adds a lot of code for a feature that isn't really worth it in my opinion.

Reply:
Wait, so we won't have typical un-obfuscate button?

Reply:
> Comptime option?

No, option in the init function

> Wait, so we won't have typical un-obfuscate button?

No, I don't see much value in it, since we won't have any passwords you would want to write by hand anyways. And IPs are copy pasted too.